### PR TITLE
Static list for dashboard params

### PR DESCRIPTION
**Pull request by qnkhuat**
The param with a static-list for dashboard will look sth like this
```diff
{
  "name": "Text starts with",
  "slug": "text_starts_with",
  "id": "bc2a98cf",
  "type": "string/=",
+ "source_type": "static-list",
+ "source_options": {"values": ["value1", "value2"]},
}
```

To retrieve the values list, call the API:
- GET /api/dashboard/:id/params/:param-key/values
- GET /api/dashboard/:id/params/:param-key/search/:query

same for embedding and public endpoint

Epic: https://github.com/metabase/metabase/issues/26873

### CODE REVIEW

File: src/metabase/util/schema.clj [hunk 1]
```diff
@@ -364,17 +364,17 @@
...
 (def Parameter
   "Schema for a valid Parameter.
   We're not using [metabase.mbql.schema/Parameter] here because this Parameter is meant to be used for
   Parameters we store on dashboard/card, and it has some difference with Parameter in MBQL."
   (with-api-error-message {:id                              NonBlankString
                            :type                            keyword-or-non-blank-str
-                           (s/optional-key :source_type)    (s/enum "static-list" "card")
-                           (s/optional-key :source_options) ParameterSourceOptions
+                           (s/optional-key :source_type)    (s/enum "static-list" "card" nil)
+                           (s/optional-key :source_options) Map
```
Comment (tsmacdonald) [inline src/metabase/util/schema.clj:377]: I agree with adding `nil` to `source_type`, but why the other changes?

Comment (tsmacdonald) [reply src/metabase/util/schema.clj:377]: I'm pretty sure that `values` and `card_id` come back as keywords (not strings) from the type normalization

**Review by tsmacdonald — COMMENTED** [review]
Curious about the schema but otherwise looks good :heavy_check_mark:

File: src/metabase/util/schema.clj (hunk 1 above)
Comment (qnkhuat) [reply src/metabase/util/schema.clj:377]: yep it is, I changed it to a keyword and use the paramtersourceoptions

**Pull request merged**